sdk/openfeature-web-provider/src/DevCycleProvider.ts

```diff
@@ -0,0 +1,351 @@
+import {
+    Provider,
+    ResolutionDetails,
+    EvaluationContext,
+    JsonValue,
+    EvaluationContextValue,
+    ProviderMetadata,
+    StandardResolutionReasons,
+    ParseError,
+    TargetingKeyMissingError,
+    InvalidContextError,
+    ProviderStatus,
+} from '@openfeature/web-sdk'
+import {
+    initializeDevCycle,
+    DevCycleClient,
+    DevCycleOptions,
+    DVCVariable,
+    DevCycleUser,
+    DVCJSON,
+    DVCCustomDataJSON,
+} from '@devcycle/js-client-sdk'
+import { VariableValue } from '@devcycle/types'
+
+const DVCKnownPropertyKeyTypes: Record<string, string> = {
+    email: 'string',
+    name: 'string',
+    language: 'string',
+    country: 'string',
+    appVersion: 'string',
+    appBuild: 'number',
+    customData: 'object',
+    privateCustomData: 'object',
+}
+
+type EvaluationContextObject = {
+    [key: string]: EvaluationContextValue
+}
+
+export default class DevCycleProvider implements Provider {
+    readonly metadata: ProviderMetadata = {
+        name: 'devcycle-web-provider',
+    } as const
+
+    readonly runsOn = 'client'
+
+    private readonly options: DevCycleOptions
+    private readonly sdkKey: string
+
+    private devcycleClient: DevCycleClient | null = null
+    get DevcycleClient(): DevCycleClient | null {
```

Comment:
why should this ever be null?

Reply:
Because the `DevCycleClient` is initialized by the `initialize()` call called from the OpenFeature SDK, not the constructor.